src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java

```diff
@@ -4388,4 +4389,32 @@ && isInternalCase(asylumCase)
         );
     }
 
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> internalDetainedAppealFeeDueNotificationHandler(
+            @Qualifier("internalDetainedAppealFeeDueNotificationGenerator") List<NotificationGenerator> notificationGenerators
+    ) {
+
+        return new NotificationHandler(
+                (callbackStage, callback) -> {
+                    final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                    AppealType appealType = asylumCase.read(APPEAL_TYPE, AppealType.class)
+                            .orElseThrow(() -> new RequiredFieldMissingException("Appeal type not found"));
```

Comment:
We should take this out and revert the integration changes (we shouldn't really set it as AppealType.HU by default over there because of this), and use isEaHuEuAppeal (you won't need to update integration test if you do it this way).